.github/workflows/pull-request.yaml

```diff
@@ -9,15 +9,17 @@ jobs:
         # Test on linux with the newest version of python and pandas 1.x/2.x.
         python-version: ['3.11']
         os: [ubuntu-latest, windows-latest]
-        pandas-version: ['pandas@latest']
+        # Version lock pandas until v2.1.0 can be implemented without breaking Pyright
+        # pandas-version: ['pandas@latest']
+        pandas-version: ['pandas~2.0.3']
```

Comment:
Please use the `~=` tag for pandas 2.0 so we will test against 2.0.4 if it happens and we don't go back to floating. https://peps.python.org/pep-0440/#compatible-release